bpf/bpf_host.c

```diff
@@ -1322,23 +1321,12 @@ int cil_from_host(struct __ctx_buff *ctx)
 static __always_inline int do_encrypt_overlay(struct __ctx_buff *ctx)
 {
 	int ret = CTX_ACT_OK;
-	__u16 proto = 0;
 	struct iphdr __maybe_unused *ipv4;
 	void __maybe_unused *data, *data_end = NULL;
 
-	/* we require a valid layer 2 to proceed */
-	if (!validate_ethertype(ctx, &proto))
-		return ret;
-
-	if (proto != bpf_htons(ETH_P_IP))
-		return ret;
-
 	if (!revalidate_data(ctx, &data, &data_end, &ipv4))
 		return DROP_INVALID;
 
-	if (!vxlan_skb_is_vxlan_v4(data, data_end, ipv4, TUNNEL_PORT))
```

Comment:
At first blush I was thinking that replacing this with the overlay mark check is not equivalent, since this was determing if the traffic was vxlan. 

But I guess, internally that check is just checking if the TUNNEL_PORT matches, which in and of itself doesn't confirm anything about the tunnel proto we are using. 

I think agent side, we won't flick the ENABLE_ENCRYPTED_OVERLAY flag on if tunnel-proto is not VXLAN. So this seems okay after a bit of thought.

Reply:
Yeah I actually think this will take you a bit further towards GENEVE support. A tiny bit less of protocol parsing, no?

Reply:
Yup, looks good to me. It's cleaner just to catch traffic off the interface.